crates/core/src/runtime/stack.rs

```diff
@@ -238,3 +238,174 @@ impl<P: super::Runtime> super::Runtime for IndexFrame<P> {
         self.parent.registers()
     }
 }
+
+/// A [`StackFrame`] where variables are not recursively searched for,
+/// However, you can still access the parent's partials.
+pub struct OpaqueStackFrame<P, O> {
```

Comment:
nit: I think I'd prefer `SandboxedStackFrame` as I feel like `Opaque` isn't quite getting the point across to me

Reply:
That's fair. I couldn't really think a good name, but sandboxed works.